### DISCUSSION TITLE
Applying percentage values to named color variables

### DISCUSSION THREAD

**Discussion by woltersdylan** [davatorium/rofi#1415] [Theming] [answered]
I've been working on updating my theme to the new system. First off I do want to say that the recent changes to theming and configuration files are absolutely fantastic! Everything feels more logical, organized, and readable than ever before. So huge thanks to everyone who worked on putting all of that together. 

With that said, I am struggling to understand how to actually apply a percentage value to a named color variable. Specifically in regards to transparency. I've primarily been referencing [this](https://github.com/davatorium/rofi/blob/next/doc/rofi-theme.5.markdown#color) section of the `rofi-theme` documentation. 

Whenever I include a percentage (either 50% or 0.5) I am greeted with an error that Rofi was expecting the line-escape character `;`.  I am almost certain that I am misunderstanding the logic behind how transparency is being handled. I've looked through some of the default themes (not all of them) to find an example use as reference, but was unable to find one. 

What would a properly formatted percentage exactly look like? And, if applicable, are there any hierarchy limitations that should be factored in (I.E. Can only be applied within global configuration, or within the Window element, etc.)?

**Comment by DaveDavenport** [comment, marked as answer]
Thanks for liking the changes.  I really hoped this system would improve things when I introduced it in 2017 :D.

Its been a few years I wrote this color stuff, let me look back.

For named colors:

```css
window {
   background-color: black / 50%;
}
```

This makes the window background  black with 50% opacity.

The other formats, it is an extra optional field:
```css
window { 
    background-color: rgb(244,244,244, 50%); 
}
```
or
```css
window {
    background-color: hsl(1,0.4,0.3, 0.5); }
}
```

**Reply by woltersdylan** [reply]
Thank you, this worked perfectly! I could have sworn I attempted this, but at least I felt certain it was user error more than anything!
